systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,375 @@
+        ResourceManager.getInstance().createResourceWithoutWait(KafkaTopicTemplates.topic(
+                namespace, clusterName, topicName, numberOfPartitions, numberOfReplicas, minInSyncReplicas).build());
+    }
+
+    private static void waitForTopicStatus(String namespace, String topicName, Enum<?> conditionType, ConditionStatus conditionStatus) {
+        LOGGER.debug("Waiting for KafkaTopic {}/{} to be {}", namespace, topicName, conditionType.toString());
+        KafkaTopicUtils.waitForKafkaTopicStatus(namespace, topicName, conditionType, conditionStatus);
+    }
+
+    private static void updateTopic(String namespace, String topicName, KafkaTopicSpec topicSpec) {
+        LOGGER.debug("Updating KafkaTopic {}/{}", namespace, topicName);
+        KafkaTopicResource.replaceTopicResourceInSpecificNamespace(topicName, kafkaTopic -> kafkaTopic.setSpec(topicSpec), namespace);
+    }
+
+    public static void waitForTopicConfigContains(String namespace, String topicName, Map<String, Object> config) {
+        LOGGER.debug("Waiting for KafkaTopic: {}/{} to contain correct config", namespace, topicName);
+        TestUtils.waitFor("KafkaTopic: " + namespace + "/" + topicName + " to contain correct config",
+                TestConstants.GLOBAL_POLL_INTERVAL, TestConstants.GLOBAL_STATUS_TIMEOUT,
+                () -> KafkaTopicUtils.configsAreEqual(KafkaTopicResource.kafkaTopicClient()
+                        .inNamespace(namespace).withName(topicName).get().getSpec().getConfig(), config)
+        );
+        LOGGER.debug("KafkaTopic: {}/{} contains correct config", namespace, topicName);
+    }
+
+    private static void deleteKafkaTopic(String namespace, String topicName) {
+        LOGGER.debug("Deleting topic {}", topicName);
+        ResourceManager.cmdKubeClient().namespace(namespace).deleteByName(KafkaTopic.RESOURCE_SINGULAR, topicName);
+    }
+
+    public static void waitForKafkaTopicDeletion(String namespaceName, String topicName) {
+        LOGGER.debug("Waiting for KafkaTopic: {}/{} deletion", namespaceName, topicName);
+        TestUtils.waitFor("deletion of KafkaTopic: " + namespaceName + "/" + topicName,
+                TestConstants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+                TestConstants.GLOBAL_TIMEOUT_SHORT,
+                () -> {
+                    if (KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicName).get() == null) {
+                        return true;
+                    } else {
+                        LOGGER.warn("KafkaTopic: {}/{} is not deleted yet! Triggering force delete by cmd client!", namespaceName, topicName);
+                        cmdKubeClient(namespaceName).deleteByName(KafkaTopic.RESOURCE_KIND, topicName);
+                        return false;
+                    }
+                },
+                () -> LOGGER.info(KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicName).get())
+        );
+    }
+}
```

Comment:
Aren't some of the methods already implemented in the `KafkaTopicUtils` or something like that? I would move it outside of the "performance utils" as they can be (IMHO) used somewhere else

Reply:
It was part of Federico's aux methods but I then realised that they are not needed cause everything can be replaced by
```java
/**
     * Executes the full lifecycle of Topic Operator tasks which includes creation, modification,
     * and deletion operations. These operations are encapsulated as a single task that is suitable
     * for parallel processing.
     *
     * @param topicIndex        the index of the topic which identifies the specific topic to be managed.
     * @param testStorage       an object representing the storage where test data or states are maintained.
     * @param extensionContext  an object representing current context of the test case
     */
    private static void performFullLifecycle(int topicIndex, TestStorage testStorage, ExtensionContext extensionContext) {
        performCreationWithWait(topicIndex, topicIndex + 1, extensionContext, testStorage);
        performModificationWithWait(topicIndex, topicIndex + 1, extensionContext, testStorage, KAFKA_TOPIC_CONFIG_TO_MODIFY);
        performDeletionWithWait(topicIndex, topicIndex + 1, extensionContext, testStorage);
    }
```
and so I have replaced it with that. So this method will be not here.